packages/stream-tile-handler/src/schema-utils.ts

```diff
@@ -30,10 +35,14 @@ export class SchemaValidation {
     schemaStreamId: string
   ): Promise<void> {
     const schema = await this._loadSchemaById(ceramic, schemaStreamId)
+    if (!isRecord(schema)) throw new Error(`Invalid schema ${schemaStreamId}: Not object`)
```

Comment:
I don't really understand this - what situation is this trying to catch/prevent?

Reply:
Whatever is referenced by `schemaStreamId` is free to contain some garbage, which is definitely not a schema, like a single string.

Reply:
ah, so you make a TileDocument, put some random data in it that isn't actually a json-schema, then use that TileDocument's streamid as the schema field for another tile.  This is the first place where we'd notice.  Makes sense

Reply:
Sarcasm? :) Types could be a pain, but in general, if we do not try to overstep limits of a particular type system, they highlight risks like that.

Regarding overstepping: TypeScript types are definitely not enough to fully implement dependent types (="this array contains strictly N elements based on this constructor parameter `n`"), or [algebraic effects](https://overreacted.io/algebraic-effects-for-the-rest-of-us/). Fun fact: vanilla JS is better suited for algebraic effects. Some time here and there we would reach the limits. Mostly though, TS types are expressive enough.

Reply:
Nope, I wasn't being sarcastic!  I was genuinely trying to understand in what situation someone could trigger that error.  :slightly_smiling_face: